src/game/client/worldheightmap.cpp

```diff
@@ -29,21 +45,2359 @@ class GDIFileStream : public InputStream
+
+        for (int i = 0; i < m_height; i++) {
+            for (int j = 0; j < w; j++) {
+                m_cliffState[j + m_flipStateWidth * i] = c[j + w * i];
+            }
+        }
+    } else {
+        file.Read_Byte_Array(m_cliffState, m_flipStateWidth * m_height);
+    }
+
+    m_numBitmapTiles = file.Read_Int32();
+    captainslog_dbgassert(m_numBitmapTiles > 0 || m_numBitmapTiles < 2048, "Unlikely numBitmapTiles.");
+    m_numBlendedTiles = file.Read_Int32();
+    captainslog_dbgassert(m_numBlendedTiles > 0 || m_numBlendedTiles < 16193, "Unlikely numBlendedTiles.");
+
+    if (info->version < 5) {
+        m_numCliffInfo = 1;
+    } else {
+        m_numCliffInfo = file.Read_Int32();
+    }
+
+    m_numTextureClasses = file.Read_Int32();
+    captainslog_dbgassert(m_numTextureClasses > 0 || m_numTextureClasses < 16193, "Unlikely m_numTextureClasses.");
+
+    for (int i = 0; i < m_numTextureClasses; i++) {
+        m_textureClasses[i].global_texture_class = 0xFFFFFFFF;
+        m_textureClasses[i].first_tile = file.Read_Int32();
+        m_textureClasses[i].num_tiles = file.Read_Int32();
+        m_textureClasses[i].width = file.Read_Int32();
+        file.Read_Int32();
+        m_textureClasses[i].name = file.Read_AsciiString();
+        Read_Tex_Class(&m_textureClasses[i], m_sourceTiles);
+    }
+
+    m_numEdgeTextureClasses = 0;
+    m_numEdgeTiles = 0;
+
+    if (info->version >= 4) {
+        m_numEdgeTiles = file.Read_Int32();
+        m_numEdgeTextureClasses = file.Read_Int32();
+
+        for (int i = 0; i < m_numEdgeTextureClasses; i++) {
+            m_edgeTextureClasses[i].global_texture_class = 0xFFFFFFFF;
+            m_edgeTextureClasses[i].first_tile = file.Read_Int32();
+            m_edgeTextureClasses[i].num_tiles = file.Read_Int32();
+            m_edgeTextureClasses[i].width = file.Read_Int32();
+            m_edgeTextureClasses[i].name = file.Read_AsciiString();
+            Read_Tex_Class(&m_edgeTextureClasses[i], m_edgeTiles);
+        }
+    }
+
+    for (int i = 1; i < m_numBlendedTiles; i++) {
+        m_blendedTiles[i].blend_ndx = file.Read_Int32();
+        m_blendedTiles[i].horiz = file.Read_Byte();
+        m_blendedTiles[i].vert = file.Read_Byte();
+        m_blendedTiles[i].right_diagonal = file.Read_Byte();
+        m_blendedTiles[i].left_diagonal = file.Read_Byte();
+        m_blendedTiles[i].inverted = file.Read_Byte();
+
+        if (!g_theWriteableGlobalData->m_use3WayTerrainBlends) {
+            m_blendedTiles[i].inverted &= ~2;
+        }
+
+        if (info->version < 3) {
+            m_blendedTiles[i].long_diagonal = 0;
+        } else {
+            m_blendedTiles[i].long_diagonal = file.Read_Byte();
+        }
+
+        if (info->version < 4) {
+            m_blendedTiles[i].custom_blend_edge_class = 0xFFFFFFFF;
+        } else {
+            m_blendedTiles[i].custom_blend_edge_class = file.Read_Int32();
+        }
+
+        int format = file.Read_Int32();

```

Comment:
this is called flag according to bfme2 and the value is FLAG_VAL

Reply:
fixed.